parlai/core/logs.py

```diff
@@ -68,6 +69,8 @@ def log_metrics(self, setting, step, report):
         for k, v in report.items():
             if isinstance(v, numbers.Number):
                 self.writer.add_scalar(f'{setting}/{k}', v, global_step=step)
+            elif isinstance(v, Metric):
```

Comment:
It *feels* like we shouldn't need to have a separate condition for this because it's mostly a repeat of above, but I don't know how to combine the two conditions without obfuscating what Metric does